weed/command/filer_sync.go

```diff
@@ -198,33 +212,70 @@ func doSubscribeFilerMetaChanges(clientId int32, grpcDialOption grpc.DialOption,
+	var processEventFnWithOffset pb.ProcessMetadataFunc
+
+	setOffsetFunc := func(counter int64, lastTsNs int64) error {
+		now := time.Now().UnixNano()
+		glog.V(0).Infof("sync %s to %s progressed to %v %0.2f/sec.", sourceFiler, targetFiler, time.Unix(0, lastTsNs), float64(counter)/(float64(now-lastLogTsNs)/1e9))
 		lastLogTsNs = now
 		// collect synchronous offset
 		statsCollect.FilerSyncOffsetGauge.WithLabelValues(sourceFiler.String(), targetFiler.String(), clientName, sourcePath).Set(float64(lastTsNs))
 		return setOffset(grpcDialOption, targetFiler, getSignaturePrefixByPath(sourcePath), sourceFilerSignature, lastTsNs)
-	})
+	}
+
+	if parallelNum <= 1 {
+		filerSink := &filersink.FilerSink{}
+		filerSink.DoInitialize(targetFiler.ToHttpAddress(), targetFiler.ToGrpcAddress(), targetPath, replicationStr, collection, ttlSec, diskType, grpcDialOption, sinkWriteChunkByFiler)
+		filerSink.SetSourceFiler(filerSource)
+		persistEventFn := genProcessFunction(sourcePath, targetPath, filerSink, debug)
+
+		processEventFn := func(resp *filer_pb.SubscribeMetadataResponse) error {
+			message := resp.EventNotification
+			for _, sig := range message.Signatures {
+				if sig == targetFilerSignature && targetFilerSignature != 0 {
+					fmt.Printf("%s skipping %s change to %v\n", targetFiler, sourceFiler, message)
+					return nil
+				}
+			}
+			return persistEventFn(resp)
+		}
+		processEventFnWithOffset = pb.AddOffsetFunc(processEventFn, 3*time.Second, setOffsetFunc)
+	} else {
+		glog.Infof("parallel synchronization started. parallelNum:%d, parallelMaxInterval:%s. from %s to %s. ", parallelNum, parallelMaxInterval.String(), sourceFiler.String(), targetFiler.String())
+		//  a semaphore for stopping the trigger
+		parallelSyncCancel := make(chan struct{})
+		defer func() {
+			parallelSyncCancel <- struct{}{}
+			glog.Infof("parallel synchronization exit. from %s to %s", sourceFiler.String(), targetFiler.String())
+		}()
+
+		// worker filerSink
+		var persistEventFns []func(resp *filer_pb.SubscribeMetadataResponse) error
+		for i := 0; i < parallelNum; i++ {
+			filerSinkItem := &filersink.FilerSink{}
+			filerSinkItem.DoInitialize(targetFiler.ToHttpAddress(), targetFiler.ToGrpcAddress(), targetPath, replicationStr, collection, ttlSec, diskType, grpcDialOption, sinkWriteChunkByFiler)
+			filerSinkItem.SetSourceFiler(filerSource)
+			persistEventFn := genProcessFunction(sourcePath, targetPath, filerSinkItem, debug)
+			persistEventFns = append(persistEventFns, persistEventFn)
+		}
+
+		c := ParallelSyncMetadataCache{persistEventFns: persistEventFns, l: []*filer_pb.SubscribeMetadataResponse{}, ParallelNum: parallelNum,
+			ParallelBatchSize: parallelBatchSize, sourceFiler: sourceFiler.String(), targetFiler: targetFiler.String()}
+
+		// parameter range limit
+		if c.ParallelNum < 1 {
+			c.ParallelNum = 1
+		} else if uint8(c.ParallelNum) > ^uint8(0) {
```

Comment:
what is this?

Reply:
I want to limit the range of parameters here. It's really not a good place. Is there any place to write? Or delete it？